src/store/RelayRecordWriter.js

```diff
@@ -94,7 +94,7 @@ class RelayRecordWriter {
    */
   getDataID(
     storageKey: string,
-    identifyingArgValue: ?string
+    identifyingArgValue: any
```

Comment:
I see what's happening. The type for `identifyingArgValue` has to be `mixed` here because we will now support arbitrary values, but line 106 returns the value directly which conflicts with the string return type...